Compiler Warnings in mapviz Package

When compiling mapviz with gcc -Wall -Wno-unused, I get the following warnings:

```
Warnings   << mapviz:make /home/evenator/code/dsat_catkin/logs/mapviz/build.make.008.log                                                                                                                                                     
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/mapviz_application.h:50: Note: No relevant classes found. No output generated.
In file included from /home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:31:0:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h: In constructor ‘mapviz::ServiceUpdaterThread::ServiceUpdaterThread(ros::NodeHandle&, const string&, QObject*)’:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:80:24: warning: ‘mapviz::ServiceUpdaterThread::allowed_datatype_’ will be initialized after [-Wreorder]
     const std::string& allowed_datatype_;
                        ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:69:21: warning:   base ‘QThread’ [-Wreorder]
       QThread(parent)
                     ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:66:5: warning:   when initialized here [-Wreorder]
     ServiceUpdaterThread(ros::NodeHandle& nh, const std::string& allowed_datatype, QObject* parent) :
     ^
In file included from /home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:31:0:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h: In constructor ‘mapviz::SelectServiceDialog::SelectServiceDialog(const string&, QWidget*)’:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:165:16: warning: ‘mapviz::SelectServiceDialog::name_filter_’ will be initialized after [-Wreorder]
     QLineEdit *name_filter_;
                ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:117:21: warning:   base ‘QDialog’ [-Wreorder]
       QDialog(parent)
                     ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:110:3: warning:   when initialized here [-Wreorder]
   SelectServiceDialog::SelectServiceDialog(const std::string& datatype, QWidget* parent)
   ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp: In member function ‘std::string mapviz::SelectServiceDialog::selectedService() const’:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:289:42: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (row < displayed_services_.size()) {
                                          ^
In file included from /home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/moc_select_service_dialog.cxx:9:0:
/home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/../../../../src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h: In constructor ‘mapviz::ServiceUpdaterThread::ServiceUpdaterThread(ros::NodeHandle&, const string&, QObject*)’:
/home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/../../../../src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:80:24: warning: ‘mapviz::ServiceUpdaterThread::allowed_datatype_’ will be initialized after [-Wreorder]
     const std::string& allowed_datatype_;
                        ^
/home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/../../../../src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:69:21: warning:   base ‘QThread’ [-Wreorder]
       QThread(parent)
                     ^
/home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/../../../../src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:66:5: warning:   when initialized here [-Wreorder]
     ServiceUpdaterThread(ros::NodeHandle& nh, const std::string& allowed_datatype, QObject* parent) :
     ^
In file included from /home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:31:0:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h: In constructor ‘mapviz::ServiceUpdaterThread::ServiceUpdaterThread(ros::NodeHandle&, const string&, QObject*)’:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:80:24: warning: ‘mapviz::ServiceUpdaterThread::allowed_datatype_’ will be initialized after [-Wreorder]
     const std::string& allowed_datatype_;
                        ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:69:21: warning:   base ‘QThread’ [-Wreorder]
       QThread(parent)
                     ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:66:5: warning:   when initialized here [-Wreorder]
     ServiceUpdaterThread(ros::NodeHandle& nh, const std::string& allowed_datatype, QObject* parent) :
     ^
In file included from /home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:31:0:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h: In constructor ‘mapviz::SelectServiceDialog::SelectServiceDialog(const string&, QWidget*)’:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:165:16: warning: ‘mapviz::SelectServiceDialog::name_filter_’ will be initialized after [-Wreorder]
     QLineEdit *name_filter_;
                ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:117:21: warning:   base ‘QDialog’ [-Wreorder]
       QDialog(parent)
                     ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:110:3: warning:   when initialized here [-Wreorder]
   SelectServiceDialog::SelectServiceDialog(const std::string& datatype, QWidget* parent)
   ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp: In member function ‘std::string mapviz::SelectServiceDialog::selectedService() const’:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_service_dialog.cpp:289:42: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (row < displayed_services_.size()) {
                                          ^
In file included from /home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/moc_select_service_dialog.cxx:9:0:
/home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/../../../../src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h: In constructor ‘mapviz::ServiceUpdaterThread::ServiceUpdaterThread(ros::NodeHandle&, const string&, QObject*)’:
/home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/../../../../src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:80:24: warning: ‘mapviz::ServiceUpdaterThread::allowed_datatype_’ will be initialized after [-Wreorder]
     const std::string& allowed_datatype_;
                        ^
/home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/../../../../src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:69:21: warning:   base ‘QThread’ [-Wreorder]
       QThread(parent)
                     ^
/home/evenator/code/dsat_catkin/build/mapviz/include/mapviz/../../../../src/MARTI/mapviz/mapviz/include/mapviz/select_service_dialog.h:66:5: warning:   when initialized here [-Wreorder]
     ServiceUpdaterThread(ros::NodeHandle& nh, const std::string& allowed_datatype, QObject* parent) :
     ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_topic_dialog.cpp: In member function ‘std::vector<ros::master::TopicInfo> mapviz::SelectTopicDialog::selectedTopics() const’:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_topic_dialog.cpp:199:44: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t i = 0; i < qt_selection.size(); i++) {
                                            ^
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_topic_dialog.cpp: In member function ‘std::vector<ros::master::TopicInfo> mapviz::SelectTopicDialog::selectedTopics() const’:
/home/evenator/code/dsat_catkin/src/MARTI/mapviz/mapviz/src/select_topic_dialog.cpp:199:44: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t i = 0; i < qt_selection.size(); i++) {
                                            ^
cd /home/evenator/code/dsat_catkin/build/mapviz; catkin build --get-env mapviz | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -
```

--- a/mapviz/include/mapviz/select_service_dialog.h
+++ b/mapviz/include/mapviz/select_service_dialog.h
@@ -64,9 +64,9 @@ namespace mapviz
     Q_OBJECT
   public:
     ServiceUpdaterThread(ros::NodeHandle& nh, const std::string& allowed_datatype, QObject* parent) :
+      QThread(parent),
       nh_(nh),
-      allowed_datatype_(allowed_datatype),
-      QThread(parent)
+      allowed_datatype_(allowed_datatype)
     {
     }
     void run();
@@ -150,19 +150,19 @@ namespace mapviz
     void timerEvent(QTimerEvent *);
     void closeEvent(QCloseEvent *);
 
-    std::string allowed_datatype_;
-
     ros::NodeHandle nh_;
 
-    std::vector<std::string> known_services_;
+    std::string allowed_datatype_;
     std::vector<std::string> displayed_services_;
+    std::vector<std::string> known_services_;
 
     int fetch_services_timer_id_;
 
-    QPushButton *ok_button_;
     QPushButton *cancel_button_;
     QListWidget *list_widget_;
     QLineEdit *name_filter_;
+    QPushButton *ok_button_;
+
     boost::shared_ptr<ServiceUpdaterThread> worker_thread_;
   };
 }

--- a/mapviz/src/select_service_dialog.cpp
+++ b/mapviz/src/select_service_dialog.cpp
@@ -109,12 +109,12 @@ namespace mapviz
 
   SelectServiceDialog::SelectServiceDialog(const std::string& datatype, QWidget* parent)
       :
+      QDialog(parent),
       allowed_datatype_(datatype),
-      ok_button_(new QPushButton("&Ok")),
       cancel_button_(new QPushButton("&Cancel")),
       list_widget_(new QListWidget()),
       name_filter_(new QLineEdit()),
-      QDialog(parent)
+      ok_button_(new QPushButton("&Ok"))
   {
     QHBoxLayout *filter_box = new QHBoxLayout();
     filter_box->addWidget(new QLabel("Filter:"));
@@ -286,7 +286,7 @@ namespace mapviz
 
     if (qt_selection.isValid()) {
       int row = qt_selection.row();
-      if (row < displayed_services_.size()) {
+      if (row < static_cast<int>(displayed_services_.size())) {
         return displayed_services_[row];
       }
     }

--- a/mapviz/src/select_topic_dialog.cpp
+++ b/mapviz/src/select_topic_dialog.cpp
@@ -196,7 +196,7 @@ std::vector<ros::master::TopicInfo> SelectTopicDialog::selectedTopics() const
 
   std::vector<ros::master::TopicInfo> selection;
   selection.resize(qt_selection.size());
-  for (size_t i = 0; i < qt_selection.size(); i++) {
+  for (int i = 0; i < qt_selection.size(); i++) {
     if (!qt_selection[i].isValid()) {
       continue;
     }

--- a/mapviz_plugins/src/point_drawing_plugin.cpp
+++ b/mapviz_plugins/src/point_drawing_plugin.cpp
@@ -300,7 +300,7 @@ namespace mapviz_plugins
     transformed = transformed | TransformPoint(cur_point_);
     if (laps_.size() > 0)
     {
-      for (int i = 0; i < laps_.size(); i++)
+      for (size_t i = 0; i < laps_.size(); i++)
       {
         std::list<StampedPoint>::iterator lap_it = laps_[i].begin();
         for (; lap_it != laps_[i].end(); ++lap_it)
@@ -324,7 +324,7 @@ namespace mapviz_plugins
     QColor base_color = color_;
     if (laps_.size() != 0)
     {
-      for (int i = 0; i < laps_.size(); i++)
+      for (size_t i = 0; i < laps_.size(); i++)
       {
         UpdateColor(base_color,i);
         if (draw_style_ == LINES)
@@ -403,7 +403,7 @@ namespace mapviz_plugins
     QColor base_color = color_;
     if (laps_.size() != 0)
     {
-      for (int i = 0; i < laps_.size(); i++)
+      for (size_t i = 0; i < laps_.size(); i++)
       {
         UpdateColor(base_color,i);
         std::list<StampedPoint>::iterator it = laps_[i].begin();

